kernel/src/process.rs

```diff
@@ -1619,10 +1627,45 @@ impl<C: 'static + Chip> Process<'_, C> {
         // Minimum memory size for the process.
         let min_total_memory_size = min_app_ram_size + initial_kernel_memory_size;
 
+        // Check if this process requires a fixed memory start address. If so,
+        // try to adjust the memory region to work for this process.
+        //
+        // Right now, we only support skipping some RAM and leaving a chunk
+        // unused so that the memory region starts where the process needs it
+        // to.
+        let (new_memory_start, new_memory_size) =
+            if let Some(fixed_memory_start) = tbf_header.get_fixed_address_ram() {
+                // The process does have a fixed address. See if it comes after
+                // where we are in memory.
+                if fixed_memory_start > remaining_app_memory as u32 {
+                    let diff: usize = (fixed_memory_start - remaining_app_memory as u32) as usize;
+                    if diff > remaining_app_memory_size {
+                        // We ran out of memory. Return the original range since (in
+                        // this step) we were unable to meet the process's
+                        // requirements.
+                        (remaining_app_memory, remaining_app_memory_size)
```

Comment:
This is confusing. Why would this not return an error here?

Reply:
My thought was that this step is very narrowly defined in the scope of what it can do. However, perhaps the MPU step could figure something better out that would support the application.